Add gravity compensation filter (#153)
ros-controls/control_toolbox: control_toolbox/src/control_filters/README.md

## Available filters

* Gravity Compensation: implements a gravity compensation algorithm, removing the gravity component from the incoming data (Wrench).
* Low Pass: implements a low-pass filter based on a time-invariant [Infinite Impulse Response (IIR) filter](https://en.wikipedia.org/wiki/Infinite_impulse_response), for different data types (doubles or wrench).
* Exponential Filter: Exponential filter for double data type.

## Gravity compensation filter

>  This filter implements a gravity compensation algorithm, applied to an `data_in` wrench, computed at a `sensor_frame` in which the center of gravity (CoG) of the to-be-compensated mass is known.

Author: Hi, as the original author of the PR, I still get the notifications, and although I am not into ROS 2 anymore, and don't know if changes to clock constructor (ROS or SYSTEM times) and others are required, I still can report on english text change.
if we simplify the sentence we see that the suggested fix does now work grammatically.
see "This filter implements an algorithm, applied to a wrench, computed at a frame in which the center of gravity (CoG) known mass." 
I can understand that sentence is not simple, so what about rephrasing :

This filter implements an algorithm compensating for the gravity forces acting at the center of gravity (CoG) of a known mass, computed at a `sensor_frame` and applied to a `data_in` wrench.

>  This filter implements a gravity compensation algorithm, applied to an `data_in` wrench, computed at a `sensor_frame` in which the center of gravity (CoG) of the to-be-compensated mass is known.

Reviewer: Sounds fine to me :)

>  This filter implements a gravity compensation algorithm, applied to an `data_in` wrench, computed at a `sensor_frame` in which the center of gravity (CoG) of the to-be-compensated mass is known.

Reviewer: ROS time would make more sense because when the use_sim_time parameter is not set, it is nothing but SYSTEM time and when the parameter is set, it uses the simulation clock

>  This filter implements a gravity compensation algorithm, applied to an `data_in` wrench, computed at a `sensor_frame` in which the center of gravity (CoG) of the to-be-compensated mass is known.

Author: Here the clock is only used to throttle the messages for a human to read a warnings/errors in a slow fashion (instead of maybe 1 kHz of a controller update, so following the sim time makes no sense. Suppose you have a 10 times faster than realtime sim, this means the console would be flooded 10 times more than expected, by this 5 second throttled messages. I think SYSTEM time is what I aimed for due to logs making sense at human rate and human rate is SYSTEM time. I would agree with ROS time if there were any computation related to the clock, which is not the case as far as I can see where the clock_ is used.

>  This filter implements a gravity compensation algorithm, applied to an `data_in` wrench, computed at a `sensor_frame` in which the center of gravity (CoG) of the to-be-compensated mass is known.

Author: You are right, I did a search for clock_ but missed the one passed to tfBuffer, it was right below your change, really no idea why I missed it. 2 years and I forgot things.  Yes tf::transforms require the same clock as the transforms so ROS time is even mandatory. Probably due to a powerful computer I had at the time, that was running gazebo in realtime, I did not notice a the problem.
So good catch, thanks.

>  This filter implements a gravity compensation algorithm, applied to an `data_in` wrench, computed at a `sensor_frame` in which the center of gravity (CoG) of the to-be-compensated mass is known.

Reviewer 2: ```suggestion
This filter implements an algorithm compensating for the gravity forces acting at the center of gravity (CoG) of a known mass, computed at a `sensor_frame` and applied to a `data_in` wrench.
```
